src/main/java/dev/binclub/javaception/klass/Klass.java

```diff
@@ -71,4 +101,27 @@ public MethodInfo findMethod(String name, Type[] descriptor) {
 		}
 		return null;
 	}
+	
+	public int getMethodID(String name, String descriptor) {
+		for (int i = 0; i < methods.length; i++) {
+			MethodInfo method = methods[i];
+			if (name.equals(method.name) && method.signature.equals(descriptor)) {
+				return i;
+			}
+		}
+		throw new ClassFormatError("Could not find method id for " + name);
+	}
+	
+	public void resolve() {
+		resolved = true;
+		for (MethodInfo method : methods) {
+			if (method.name.equals("<clinit>")) {
+				ExecutionEngine.invokeMethodObj(null, method);
```

Comment:
Initialisation is the process of calling <clinit> which happens at a different time to linking/resolving (JVMS 5).

Reply:
this is called by the execution engine to ensure the static fields have ran